### DISCUSSION TITLE
What is electricity and why do I need it?

### DISCUSSION THREAD

**Discussion by Katkee29** [rapid7/metasploit-framework#15692] [Q&A] [answered]
Why does my session die after i turn off my 2nd machine that i installed malware ( meterpreter ) and is there anyway to set that malware runs again when pc is running again, thanks

**Comment by h00die** [comment, marked as answer]
Without further information about what exploit you used, you're most likely running the payload in memory or from a file. Rebooting will stop that process without restarting it. You're looking for a persistence mechanism. Metasploit has quite a few depending on the victim OS.

For instance if your victim is Linux based, check in `exploits/linux/local/` for persistence modules. Change out Linux for other OSes to fit your victim

**Reply by Katkee29** [reply]
Im using in metasploit meterpreter/reverse_tcp im trying it on windows 10 and everytime machine shuts down malware dies

**Reply by bcoles** [reply]
Various modules exist to achieve persistence:

```
$ find modules/ -name *persistence* 
modules/exploits/unix/local/at_persistence.rb
modules/exploits/osx/local/persistence.rb
modules/exploits/windows/local/persistence.rb
modules/exploits/windows/local/persistence_image_exec_options.rb
modules/exploits/windows/local/registry_persistence.rb
modules/exploits/windows/local/s4u_persistence.rb
modules/exploits/windows/local/persistence_service.rb
modules/exploits/windows/local/vss_persistence.rb
modules/exploits/windows/local/wmi_persistence.rb
modules/exploits/linux/local/apt_package_manager_persistence.rb
modules/exploits/linux/local/autostart_persistence.rb
modules/exploits/linux/local/rc_local_persistence.rb
modules/exploits/linux/local/service_persistence.rb
modules/exploits/linux/local/bash_profile_persistence.rb
modules/exploits/linux/local/cron_persistence.rb
modules/exploits/linux/local/yum_package_manager_persistence.rb
modules/post/windows/manage/persistence_exe.rb
modules/post/windows/manage/sshkey_persistence.rb
modules/post/linux/manage/sshkey_persistence.rb
```
